app/controllers/chargeback_controller.rb

```diff
@@ -573,18 +685,38 @@ def cb_rate_get_form_vars
     @sb[:rate] = @edit[:rate]
     @edit[:new][:description] = params[:description] if params[:description]
     @edit[:new][:details].each_with_index do |_detail, i|
-      @edit[:new][:details][i][:rate] = params["rate_#{i}".to_sym] if params["rate_#{i}".to_sym]
       @edit[:new][:details][i][:per_time] = params["per_time_#{i}".to_sym] if params["per_time_#{i}".to_sym]
       @edit[:new][:details][i][:per_unit] = params["per_unit_#{i}".to_sym] if params["per_unit_#{i}".to_sym]
+      # Save tiers into @edit
+      (0..@sb[:num_tiers][i].to_i - 1).each do |j|
+        @edit [:new][:tiers][i][j] = {} unless @edit[:new][:tiers][i][j]
+        @edit[:new][:tiers][i][j][:fix_rate] =
+          params["fix_rate_#{i}_#{j}".to_sym] if params["fix_rate_#{i}_#{j}".to_sym]
+        @edit[:new][:tiers][i][j][:var_rate] =
+          params["var_rate_#{i}_#{j}".to_sym] if params["var_rate_#{i}_#{j}".to_sym]
+        @edit[:new][:tiers][i][j][:start] = params["start_#{i}_#{j}".to_sym] if params["start_#{i}_#{j}".to_sym]
+        @edit[:new][:tiers][i][j][:end] = params["end_#{i}_#{j}".to_sym] if params["end_#{i}_#{j}".to_sym]
```

Comment:
The controller is checking which field has been updated in the view looking in the params hash and updating the @edit variable correspondingly.